History/js/new.js

```diff
@@ -249,6 +246,34 @@ var callbacks = {
       // Clear the old iframe and insert the new one
       $(".privly_iframe").empty();
       $(".privly_iframe").append(iFrame);
+    }
+
+    $('button#prev_preview').on('click',function(){
+      //finds the previous button with preview_link class and passes its reference.
+      //a small example of why this works : http://jsfiddle.net/5QdgB/
+      var index = $('.preview_link').index(buttonClicked);
+      var previous = $('.preview_link').slice(index-1).first();
+      console.log("previous = "+previous);
+      buttonClicked = previous;
+      iframeReturn(previous);
+    });
+
+    $('button#next_preview').on('click',function(){
+      //finds the next button with preview_link class and passes its reference.
+      var index = $('.preview_link').index(buttonClicked);
+      if(index == $('.preview_link').length - 1){
+        index = -1;     //rolling back to the first message after the last.
+      }
+      var next = $('.preview_link').slice(index+1).first();
+      console.log("next = "+next);
+      buttonClicked=next;
+      iframeReturn(next);
+    });
+    
+    $('button.preview_link').on('click', function() {      
+      buttonClicked = $(this);
```

Comment:
Unless I am mistaken, I don't see a declaration for several of these variables which would put them in the global space.